### PR TITLE
Document how to add autocompletion

### LINKED ISSUE #9: Command autocomplete
Adds implementation for autocompleting command names as suggested by @akrabat

### PR DESCRIPTION
**Pull request by weierophinney**
Details how to add autocompletion for laminas-cli, via bamarni/symfony-console-autocomplete.

Fixes #9

### CODE REVIEW

File: docs/book/autocompletion.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+- For ZSH users, `$HOME/.zshrc`
+- For FISH users, `$HOME/.config/fish/config.fish`
+
+Once the changes have been made, source your shell configuration:
+
+- For BASH users, `source $HOME/.bashrc`
+- For ZSH users, `source $HOME/.zshrc`
+- For FISH users, `source $HOME/.config/fish/config.fish`
+
+At this point, when you are in a project that uses laminas-cli, you can type:
```
Comment (michalbundyra) [inline docs/book/autocompletion.md:48]: ... after restarting shell :)

**Comment by weierophinney** [comment]
@michalbundyra That's why I mention the usage of `source` at the end of the document - that applies it to the current shell.

File: docs/book/autocompletion.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# Autocompletion
+
+laminas-cli does not provide autocompletion out of the box. However, it is
+possible to add autocompletion via a third-party package,
+[bamarni/symfony-console-autocomplete](https://github.com/bamarni/symfony-console-autocomplete).
+This package provides completion for any symfony/console-based application,
+which means it will work with laminas-cli as well.
+
+First, install the autocompletion package as a global tool:
```
Comment (froschdesign) [inline docs/book/autocompletion.md:9]:
```suggestion
## Installation

First, install the autocompletion package as a global tool:
```

File: docs/book/autocompletion.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+```bash
+composer global require bamarni/symfony-console-autocomplete
+```
+
+Second, you will need to add configuration for your shell. We will add three
+things:
```
Comment (froschdesign) [inline docs/book/autocompletion.md:16]:
```suggestion
## Configuration

Second, you will need to add configuration for your shell. We will add three
things:
```

File: docs/book/autocompletion.md (hunk 1 above)
Comment (froschdesign) [reply docs/book/autocompletion.md:48]: I suggest to also add a headline like "Usage".

Comment (weierophinney) [reply docs/book/autocompletion.md]: Done, @froschdesign :)

**Pull request merged**